cpg_utils/hail_batch.py

```diff
@@ -189,12 +189,16 @@ def _process_job_attributes(
 
         self.total_job_num += 1
 
+        # Multiple jobs in the batch might reference the same attributes dict
+        # object. Avoid modifying the dict object (e.g. with pop() or update())
+        # to avoid changing the attributes of subsequently processed jobs.
+
         attributes = attributes or {}
         stage = attributes.get('stage')
         dataset = attributes.get('dataset')
         sequencing_group = attributes.get('sequencing_group')
         participant_id = attributes.get('participant_id')
-        sequencing_groups: set[str] = set(attributes.pop('sequencing_groups', []) or [])
+        sequencing_groups: set[str] = set(attributes.get('sequencing_groups') or [])
```

Comment:
Should this be `attributes.get('sequencing_groups', [])`? With this update it will raise an exception if the key `sequencing_groups` doesn't exist, whereas previously it would not.

Reply:
This reverts the line back to what it was prior to PR #183. If that key doesn't exist, it'll be `set(None or [])` which will be an empty set as expected.

Personally I'd rather write it as `… = set(attributes.get('sequencing_groups', []))` but that will newly raise an exception if someone somehow passes in `{'sequencing_groups': None}` whereas the `or []` version won't.

In summary, 🤮 🤷 …